src/common/lib/client/baserealtime.ts

```diff
@@ -231,8 +255,9 @@ class ChannelGroup {
     });
   }
 
-  subscribe(cb: (channel: string, msg: any) => void) {
-    this.subsciptions.on('message', cb);
+  async subscribe(cb: (channel: string, msg: any) => void): Promise<void> {
+    await this.join();
+    this.subscriptions.on('message', cb);
```

Comment:
If you subscribe twice then you'll join twice, and we will have duplicate processing of the presence set?

Reply:
Presence set is deduplicated, so I think this is ok